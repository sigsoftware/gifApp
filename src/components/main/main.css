.title {
    font-family: Arial, Helvetica, sans-serif;
}

body {
    padding: 60px;
}

.column1 {
    float: left;
    width: 10%;
    padding-top: 33px;
}

.column2 {
    float: left;
    width: 90%;
}

.content {
    margin-top: 20%;
}


.row:after {
    content: "";
    display: table;
    clear: both;
}

@media(max-width:550px) {
    .column1 {
        text-align: center;
        width: 80%;
    }

    .column2 {
        text-align: center;
    }
}

.button {
    border: none;
    color: white;
    padding: 15px 25px;
    text-align: center;
    border-radius: 10px;
    background-color: #0a0a23;
    font-size: 16px;
    margin: auto;
    display: block;
    margin-top: 2%;
    cursor: pointer;
    margin-bottom: 2%;
    box-shadow: 0px 0px 2px 2px rgb(0, 0, 0);
}


h1 {
    text-align: center;
    color: black;
    text-transform: uppercase;
    padding: 1px;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-weight: 80;
    position: relative;
    background: linear-gradient(to right, black, #eee, black);
}

h1::before {
    content: "";
    position: absolute;
    left: 50%;
    top: -50px;
    width: 600px;
    margin-left: -300px;
    margin-top: -220px;
    height: 600px;
    background: radial-gradient(50% 50%, ellipse closest-side, #444, black);
    z-index: -1;
}

h1 a {
    background: black;
    display: block;
    padding: 20px;
    text-decoration: none;
    letter-spacing: 30px;
    color: white;
}

.logo {
    width: 170px;
    display: block;
    margin: auto;
    margin-bottom: 3px;
}

p {
    text-align: center;
    font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
    padding-bottom: 30px;
}

.loading {
    width: 70px;
    padding: 5%;
    display: block;
    margin: auto;
}