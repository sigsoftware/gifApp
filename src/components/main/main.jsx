import React, { useState } from 'react';
import { Search } from "./../search/search";
import "./main.css";
import { Images } from '../images/images';
import { Image } from '../images/image/image';
import { History } from "./../history/history";
import { Items } from '../history/items/items';
import { useFetchGifs } from "./../customHooks/useFetchGifs";
import load from "./../assets/loading.gif";
import { useLogo } from "./../customHooks/useLogo";
import "animate.css";

const Main = () => {

    let [term, setTerm] = useState("");
    let [categories, setCategories] = useState([]);

    let { data, loading } = useFetchGifs(term);

    let startLogo = useLogo();

    let saveHistory = () => {
        if (term.trim().length >= 3) {
            setCategories([...categories, term.toLocaleUpperCase()]);
        }
    }

    return (
        <>
            <img src={startLogo?.imageUrl} alt={startLogo?.title} className="logo animate__animated animate__bounceIn" />
            <h1 className='title animate_animated animate__fadeInUp'>
                Gif App
            </h1>
            <div className='search'>
                <Search term={term} setTerm={setTerm} />
                <button onClick={saveHistory} className="button">Guardar en Historial</button>
            </div>
            <hr />
            <div className='row'>
                <div className='column1'>
                    <div className='content'>
                        <History>
                            {
                                categories.map((category, idx) => {
                                    return <Items item={category} key={idx} setTerm={setTerm} />
                                })
                            }
                        </History>
                    </div>
                </div>
                <div className='column2'>
                    {
                        (loading && term.trim().length >= 3) && <p><img src={load} alt="loadData" className='loading' /> Cargando Primera Busqueda</p>
                    }
                    {(term.trim().length >= 3 && !loading) ? <Images term={term}>
                        {
                            data.map((image, idx) => {
                                return <Image image={image} key={idx}></Image>
                            })
                        }
                    </Images> : ""}
                </div>
            </div>
        </>
    )
}

export { Main };