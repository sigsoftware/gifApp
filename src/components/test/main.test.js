
import { shallow } from "enzyme";
import { Main } from "./../main/main";

describe("Pruebas del componente Main", () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Main />);
    })

    test("Probar que la pagina se renderiza correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    })

    test("Comprobar que existe el componente Search", async () => {
        expect(wrapper.find('Search').exists()).toBe(true);
    })

    test("Comprobar que existe el componente Images", async () => {
        let defaultCategories = ["Final Fantasy VII", "Final Fantasy X"];
        wrapper = shallow(<Main defaultCategories={defaultCategories} />);
        expect(wrapper.find('Items').length).toBe(defaultCategories.length);
    })
})