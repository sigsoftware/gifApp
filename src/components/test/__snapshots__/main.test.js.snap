// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente Main Probar que la pagina se renderiza correctamente 1`] = `
<Fragment>
  <img
    className="logo animate__animated animate__bounceIn"
  />
  <h1
    className="title animate_animated animate__fadeInUp"
  >
    Gif App
  </h1>
  <div
    className="search"
  >
    <Search
      setTerm={[Function]}
      term=""
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Guardar en Historial
    </button>
  </div>
  <hr />
  <div
    className="row"
  >
    <div
      className="column1"
    >
      <div
        className="content"
      >
        <History />
      </div>
    </div>
    <div
      className="column2"
    />
  </div>
</Fragment>
`;
